DependencyInjection/AlgoliaAlgoliaSearchExtension.php

```diff
@@ -33,8 +37,40 @@ public function load(array $configs, ContainerBuilder $container)
         $container->setParameter('algolia.catch_log_exceptions', $config['catch_log_exceptions']);
         $container->setParameter('algolia.index_name_prefix', $config['index_name_prefix']);
 
-        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
-        $loader->load('services.yml');
+        // Load all bundles to auto-discover metadata
+        $bundles = $container->getParameter('kernel.bundles');
+
+        // directories for auto_detection
+        $directories = array();
+        if ($config['metadata']['auto_detection']) {
```

Comment:
It could be a good practice to make a _empty()_ test on this condition. This way we will be sure this code will never yell a _notice_ on the log.